pkg/yurthub/storage/disk/storage.go

```diff
@@ -383,6 +383,10 @@ func (ds *diskStorage) ReplaceComponentList(component string, gvr schema.GroupVe
 
 	// 2. create new file with contents
 	// TODO: if error happens, we may need retry mechanism, or add some mechanism to do consistency check.
+	// create abspath dir in case that contents is empty
+	if err := ds.fsOperator.CreateDir(absPath); err != nil {
```

Comment:
How about add the condition for verify content is empty as following:
```
if len(contents) == 0 {
    if err := ds.fsOperator.CreateDir(absPath); err != nil {
        // xxxx
    }
}
```

Reply:
ok

Reply:
we can not cover all failed cases, and I think it is reasonable to create `absPath` directory at first even if content is not empty.  so I want to take back this comment.

Reply:
done